bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeApi.java

```diff
@@ -45,33 +48,56 @@
 @NonNullByDefault
 public class SenecHomeApi {
     private static final String HTTP_PROTO_PREFIX = "http://";
+    private static final String HTTPS_PROTO_PREFIX = "https://";
 
     private final Logger logger = LoggerFactory.getLogger(SenecHomeApi.class);
     private final HttpClient httpClient;
     private final Gson gson = new Gson();
     private String hostname = "";
 
     public SenecHomeApi(HttpClient httpClient) {
-        this.httpClient = httpClient;
+        this.httpClient = configureHttpClient(httpClient);
     }
 
     public void setHostname(String hostname) {
         this.hostname = hostname;
     }
 
+    private HttpClient configureHttpClient(HttpClient httpClient) {
+        // Configure SSL context factory to trust all certificates
+        SslContextFactory.Client sslContextFactory = new SslContextFactory.Client();
+        sslContextFactory.setTrustAll(true);
+
+        // Apply SSL context factory to the HttpClient builder
+        httpClient = new HttpClient(sslContextFactory);
```

Comment:
The instanciation of the new httpClient should be moved to ``SenecHomeHandlerFactory.java`` where currently is the code to get the common one.